source/core_http_client.c

```diff
@@ -574,10 +577,23 @@ static int8_t caseInsensitiveStringCmp( const char * str1,
                                         size_t n )
 {
     size_t i = 0U;
+    /* Inclusion of inbetween variables for coverity rule 13.2 compliance */
+    int32_t firstChar;
+    int32_t secondChar;
 
     for( i = 0U; i < n; i++ )
     {
-        if( toupper( str1[ i ] ) != toupper( str2[ i ] ) )
+        /* MISRA Ref 21.13.1 [Essential type casting] */
+        /* More details at: https://github.com/FreeRTOS/coreHTTP/blob/main/MISRA.md#rule-2113 */
+        /* coverity[misra_c_2012_rule_21_13_violation] */
+        firstChar = ( int32_t ) toupper( ( ( int32_t ) str1[ i ] ) );
+
+        /* MISRA Ref 21.13.1 [Essential Type Casting] */
+        /* More details at: https://github.com/FreeRTOS/coreHTTP/blob/main/MISRA.md#rule-2113 */
+        /* coverity[misra_c_2012_rule_21_13_violation] */
+        secondChar = ( int32_t ) toupper( ( ( int32_t ) str2[ i ] ) );
```

Comment:
An int is not necessarily 32-bits. This type cast is not necessary.

Reply:
Removed the cast